webview-ui/src/components/ChatView.tsx

```diff
@@ -29,30 +43,71 @@ export const ChatView = ({
   errorMessage,
 }: ChatViewProps) => {
   const [userPrompt, setUserPrompt] = useState("");
-
   const extensionMessenger = new ExtensionMessenger();
+  let innerTextArea: HTMLTextAreaElement | null | undefined = null;
+  const endOfMessagesRef = useRef<HTMLDivElement>(null);
 
-  const handleInputOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const newval = event.target.value;
-    setUserPrompt(newval);
+  /**
+   * Scroll to the bottom of the chat (if messages are long enough to scroll).
+   */
+  const scrollToBottom = () => {
+    if (endOfMessagesRef.current) {
+      scrollIntoView(endOfMessagesRef.current, {
+        behavior: "smooth",
+        scrollMode: "if-needed",
+      });
+    }
   };
 
-  const handleSubmit = (
-    e:
-      | React.MouseEvent<HTMLButtonElement, MouseEvent>
-      | React.FormEvent<HTMLFormElement>
-      | React.KeyboardEvent<HTMLInputElement>,
-  ) => {
-    e.preventDefault();
+  /**
+   * Only handle sending the message if not Shift+Enter key combination.
+   */
+  const onKeyDown = (event: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    if (event.key === "Enter" && !event.shiftKey) {
+      event.preventDefault();
+      event.stopPropagation();
```

Comment:
Why are these lines important?

Reply:
Good catch. Simply a remnant from ages past. I changed it so that there is no default form submission event to prevent, so those shouldn't be needed. Will remove.